src/main/java/org/opensearch/neuralsearch/processor/chunker/ChunkerParameterParser.java

```diff
@@ -41,12 +41,16 @@ public static String parseStringParameter(final Map<String, Object> parameters,
 
     /**
      * Parse integer type parameter.
-     * Throw IllegalArgumentException if parameter is not an integer.
+     * Throw IllegalArgumentException if both parameter and default value is missing or parameter is not an integer.
      */
-    public static int parseIntegerParameter(final Map<String, Object> parameters, final String fieldName, final int defaultValue) {
+    public static int parseIntegerParameter(final Map<String, Object> parameters, final String fieldName, final Integer defaultValue) {
         if (!parameters.containsKey(fieldName)) {
-            // all integer parameters are optional
-            return defaultValue;
+            // return the default value when parameter is not existing
+            if (defaultValue == null) {
```

Comment:
I suggest splitting this method to three new methods: 
1. parseInteger: Simply parse integer with catching number format exception.
2. parseIntegerWithDefault: if field not shown, return default value, else return `parseInteger`.
3. parseIntegerWithException: if field not shown, throw IllegalStateException, else return `parseInteger`.
This won't add repeated code also can make sure the method 2 have concise semantic by making the default value type to `int`, currently default value accepts `null` which is not in a correct semantic. For runtime parameters, there isn't default value, so throwing exception is the correct semantic for them.

Reply:
Oh, I don't think the function is necessary. I have removed it.

Reply:
Since all integer parameters are optional. I will not keep the `parseIntegerWithException` method.

Reply:
I mean, runtime parameters are always available while non-runtime parameters have default values.

Reply:
Updated.